perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/ModulesYamlConfigLoader.java

```diff
@@ -0,0 +1,94 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import static cz.metacentrum.perun.core.impl.Utils.notNull;
+
+/**
+ * @author Vojtech Sassmann <vojtech.sassmann@gmail.com>
+ */
+
+public class ModulesYamlConfigLoader implements ModulesConfigLoader {
+
+	private final static Logger log = LoggerFactory.getLogger(ModulesYamlConfigLoader.class);
+
+	private static final ObjectMapper yamlObjectMapper = new ObjectMapper(new YAMLFactory());
+	private String modulesDirPath = "/etc/perun/modules/";
+
+	public ModulesYamlConfigLoader() { }
+
+	public ModulesYamlConfigLoader(String modulesDirPath) {
+		this.modulesDirPath = modulesDirPath;
+	}
+
+	@Override
+	public String loadStringProperty(String moduleName, String property) {
+		return loadPropertyNode(moduleName, property).asText();
+	}
+
+	@Override
+	public Integer loadIntegerProperty(String moduleName, String property) {
+		return loadPropertyNode(moduleName, property).isNull() ? null : loadPropertyNode(moduleName, property).asInt();
+	}
+
+	@Override
+	public List<String> loadStringListProperty(String moduleName, String property) {
+		List<String> values = new ArrayList<>();
+		JsonNode propertyNode = loadPropertyNode(moduleName, property);
+		propertyNode.iterator().forEachRemaining(node -> values.add(node.asText()));
+		return values;
+	}
+
+	@Override
+	public List<Integer> loadIntegerListProperty(String moduleName, String property) {
+		List<Integer> values = new ArrayList<>();
+		JsonNode propertyNode = loadPropertyNode(moduleName, property);
+		propertyNode.iterator().forEachRemaining(node -> values.add(node.isNull() ? null : node.asInt()));
+		return values;
+	}
+
+	private JsonNode loadPropertyNode(String moduleName, String propertyName) {
+		notNull(moduleName, "configFile");
+		notNull(propertyName, "propertyName");
+
+		String path = modulesDirPath + moduleName + ".yaml";
+
+		JsonNode root = loadModulesYamlFile(path);
+
+		JsonNode propertyNode = parsePropertyNode(root, propertyName);
+		if (propertyNode == null) {
+			throw new InternalErrorException("Failed to parse property: '" + propertyName + "' from file: " + path);
+		}
+		return propertyNode;
+	}
+
+	private static JsonNode parsePropertyNode(JsonNode root, String propertyName) {
+		JsonNode currentNode = root;
+		while (propertyName.contains(".")) {
+			String[] split = propertyName.split("\\.", 2);
+			currentNode = currentNode.get(split[0]);
+			propertyName = split[1];
+		}
+		return currentNode.get(propertyName);
+	}
+
+	private static JsonNode loadModulesYamlFile(String path) {
+		try (InputStream is = new FileInputStream(path)) {
+			return yamlObjectMapper.readTree(is);
```

Comment:
Here I would prefer this way of leading a YAML file:
```java
YAMLMapper mapper = new YAMLMapper();
return mapper.readValue(new File(path), JsonNode.class);
```

Reply:
I have updated the implementation corresponding to your suggestion.